### PR TITLE
Add `delete` subcommand to `ledger-tool bigtable`

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
When corrupted blocks are uploaded to bigtable, there should be a way to remove them easily

#### Summary of Changes
Added a `delete` subcommand to `ledger-tool bigtable` which accepts a list of slots for deletion and for each slot:
1. Fetch the confirmed block from bigtable
2. Fetch the bigtable tx info for each tx in the block
3. Check if the expected tx info (based on confirmed block) matches the tx info row
4. If the tx info matches (includes a slot check), delete the tx info
5. Delete address slot rows
6. Delete block

Also included a `--dry-run` arg which is enabled by default for safety

Fixes https://github.com/solana-labs/solana/issues/19914

### CODE REVIEW

**Review by CriesofCarrots — COMMENTED** [review]
The meat of this looks great. Just a couple comments on interface.

File: ledger-tool/src/bigtable.rs [hunk 1]
```diff
@@ -263,6 +273,31 @@ impl BigTableSubCommand for App<'_, '_> {
                                 ),
                         ),
                 )
+                .subcommand(
+                    SubCommand::with_name("delete")
```
Comment (t-nelson) [inline ledger-tool/src/bigtable.rs:277]:
```suggestion
                    SubCommand::with_name("delete-slots")
```
Let's not claim the whole namespace

File: ledger-tool/src/bigtable.rs [hunk 2]
```diff
@@ -263,6 +273,31 @@ impl BigTableSubCommand for App<'_, '_> {
...
+                        .about("Delete uploaded ledger information from BigTable")
+                        .arg(
+                                Arg::with_name("slots")
+                                    .long("slots")
+                                    .value_name("SLOTS")
+                                    .takes_value(true)
+                                    .required(true)
+                                    .index(1)
+                                    .help(
+                                        "List of comma separated slots to delete",
```
[the remark this answers is not in the document]
Comment (t-nelson) [reply ledger-tool/src/bigtable.rs:287]: And positional?

[the remark this answers is not in the document]
Comment (jstarry) [reply ledger-tool/src/bigtable.rs:287]: It already is positional with `index(1)`.. unless you I'm misunderstanding you @t-nelson

File: ledger-tool/src/bigtable.rs [hunk 3]
```diff
@@ -263,6 +273,31 @@ impl BigTableSubCommand for App<'_, '_> {
...
+                                    ),
+                                )
+                            .arg(
+                                Arg::with_name("dry_run")
+                                    .long("dry-run")
+                                    .possible_values(&["true", "false"])
+                                    .takes_value(true)
```
[the remark this answers is not in the document]
Comment (jstarry) [reply ledger-tool/src/bigtable.rs]: Ah yeah much better, thanks!

File: storage-bigtable/src/bigtable.rs [hunk 4]
```diff
@@ -226,6 +229,31 @@ impl BigTableConnection {
         .await
     }
 
+    pub async fn delete_rows_with_retry(&self, table: &str, row_keys: &[RowKey]) -> Result<()> {
+        use backoff::{future::retry, ExponentialBackoff};
```
[the remark this answers is not in the document]
Comment (jstarry) [reply storage-bigtable/src/bigtable.rs]: Nope, pulled those out

File: ledger-tool/src/bigtable.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (t-nelson) [reply ledger-tool/src/bigtable.rs]: Ah sorry I stopped reading at `long`, thought that conflicted with `index`.  What's the value of specifying both here?

[the remark this answers is not in the document]
Comment (jstarry) [reply ledger-tool/src/bigtable.rs]: No value, I had forgotten to remove `index(1)` is all haha. I removed the `long` and kept `index` in the latest patch so we should be all good here

**Review by t-nelson — APPROVED** [review]

File: storage-bigtable/src/bigtable.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply storage-bigtable/src/bigtable.rs]: Thanks!

**Review by CriesofCarrots — APPROVED** [review]
lgtm

**Pull request merged**